common/darwin/Classes/FlutterRTCPeerConnection.m

```diff
@@ -506,19 +506,18 @@ - (void)peerConnection:(RTCPeerConnection*)peerConnection didOpenDataChannel:(RT
     dataChannel.eventChannel = eventChannel;
     dataChannel.flutterChannelId = flutterChannelId;
 
-    FlutterEventSink eventSink = peerConnection.eventSink;
-    if(eventSink){
-        eventSink(@{
-                    @"event" : @"didOpenDataChannel",
-                    @"id": dataChannelId,
-                    @"label": dataChannel.label,
-                    @"flutterId": flutterChannelId
-                    });
-    }
-
     dispatch_async(dispatch_get_main_queue(), ^{
        // setStreamHandler on main thread
        [eventChannel setStreamHandler:dataChannel];
+       FlutterEventSink eventSink = peerConnection.eventSink;
+       if(eventSink){
+           eventSink(@{
+                       @"event" : @"didOpenDataChannel",
+                       @"id": dataChannelId,
+                       @"label": dataChannel.label,
+                       @"flutterId": flutterChannelId
+                       });
+       }
```

Comment:
What is the reason to move `eventSink` call into `dispatch_async`?

Reply:
Oh sorry this wasn't supposed to be here. It's from another pull request https://github.com/flutter-webrtc/flutter-webrtc/pull/975.

Should I remove this commit from here?

Reply:
Yes, it will be great. Mixing different pull requests in one is not good.

Reply:
Ok, removed!